bloom-housing/bloom
# 1388/updates to date component (breaking change)

# Pull Request Template

## Issue

Addresses #1388

## Description

The only date field we had in the app was `DOBField` - but we need multiple date fields in Listings Management for things like application due date, postmark due date, event dates, etc. This ticket pulls any birthdate logic out of being default in the component into a new prop on the component (validateAge18) and renames the field to just be DateField. Confirmed with Jesse the places where we do wanted to be validating 18 and over and the places where we don't (validating for account creation and primary applicant in both public and partners, but not for household members).

## Type of change


## How Can This Be Tested/Reviewed?

You can check out the stories and the places in the app where we use a birthdate field, which include account creation, account editing, primary applicant during application process, household applicant during application process. We don't currently have a vanilla date field but we will in the next few PRs as we build the Listings Management form out.

## Checklist:
